resources/bolt_bash_completion.sh

```diff
@@ -6,134 +6,171 @@
+        esac
+      fi
+      ;;
+    group)
+      if [ $COMP_CWORD -eq 2 ]; then
+        next="show"
+      elif [ $action == 'show' ]; then
+        next="--project --format --inventoryfile"
+      fi
+      ;;
+    guide)
+      next="--format"
+      ;;
+    inventory)
+      if [ $COMP_CWORD -eq 2 ]; then
+        next="show"
+      elif [ $action == 'show' ]; then
+        next="${inventory_opts} --project --format --inventoryfile --detail"
+      fi
+      ;;
+    lookup)
+      next="${action_opts} --hiera-config"
+      ;;
+    module)
+      if [ $COMP_CWORD -eq 2 ]; then
+        next="add generate-types install show"
+      else
+        case $action in
+          add | install) next="--project";;
+          generate-types) next="${context_opts}";;
+          show) next="${context_opts} --filter --format";;
+        esac
+      fi
+      ;;
+    plan)
+      if [ $COMP_CWORD -eq 2 ]; then
+        next="convert new run show"
+      elif [[ $COMP_CWORD -eq 3 && $action != "new" ]]; then
+        if [ -f "${PWD}/.plan_cache.json" ]; then
+          # Use Puppet's ruby instead of jq since we know it will be there.
+          next=$(get_json_keys "${PWD}/.plan_cache.json")
+        elif [ -f "${HOME}/.puppetlabs/etc/bolt/.plan_cache.json" ]; then
+          next=$(get_json_keys "${HOME}/.puppetlabs/etc/bolt/.plan_cache.json")
+        fi
+      else
+        case $action in
+          convert) next="${context_opts}";;
+          new) next="--project --pp";;
+          run) next="${action_opts} ${apply_opts} --params --tmpdir";;
+          show) next="${context_opts} --filter --format";;
+        esac
+      fi
+      ;;
+    project)
+      if [ $COMP_CWORD -eq 2 ]; then
+        next="init migrate"
+      else
+        case $action in
+          init) next="--modules";;
+          migrate) next="--project --inventoryfile";;
+        esac
+      fi
+      ;;
+    script)
+      if [ $COMP_CWORD -eq 2 ]; then
+        next="run"
+      elif [[ $COMP_CWORD -eq 3 && $action == 'run' ]]; then
+        # List files and directories
+        next=$(compgen -f -d "" -- $cur)
+      elif [ $action == 'run' ]; then
+        next="${action_opts} --env-var --tmpdir"
+      fi
+      ;;
+    secret)
+      if [ $COMP_CWORD -eq 2 ]; then
+        next="createkeys encrypt decrypt"
+      else
+        case $action in
+          createkeys) next="${context_opts} --plugin --force";;
+          encrypt | decrypt) next="${context_opts} --plugin";;
+        esac
+      fi
+      ;;
+    task)
+      if [ $COMP_CWORD -eq 2 ]; then
+        next="run show"
+      elif [ $COMP_CWORD -gt 2 ]; then
+        if [ -f "${PWD}/.task_cache.json" ]; then
+          # Use Puppet's ruby instead of jq since we know it will be there.
+          next=$(get_json_keys "${PWD}/.task_cache.json")
+        elif [ -f "${HOME}/.puppetlabs/etc/bolt/.task_cache.json" ]; then
+          next=$(get_json_keys "${HOME}/.puppetlabs/etc/bolt/.task_cache.json")
+        fi
```

Comment:
Not sure if there's anything that can be done about this, but if the cache file for either plans or tasks doesn't exist then bash will try to autocomplete with the contents of the cwd:

![image](https://user-images.githubusercontent.com/30373575/121409035-e34e7c80-c915-11eb-87ed-825d8045017a.png)

Reply:
I updated this to show flag options if the tasks aren't found - I don't want to show nothing, since I'm afraid folks will think that the tab-complete stopped working or something, but if there's a better thing to do (maybe print a message below or something about needing to run `bolt module generate-types` to tab complete task names?) I can try to make it happen.

Reply:
I think just falling back to listing options is enough, especially since it seems like most users won't run into missing cache files frequently (especially considering the plan cache is generated whenever a module is installed).